tests/image_test.py

```diff
@@ -1225,5 +1226,29 @@ def test_load_img(self, tmpdir):
                                        interpolation="unsupported")
 
 
+    def test_list_pictures(self, tmpdir):
+        os.mkdir(os.path.join(tmpdir, 'class1'))
+        os.mkdir(os.path.join(tmpdir, 'class2'))
+
+        filenames = ['test.png', 'test0.jpg', 'test-1.jpeg', '2test.bmp',
+                     '2-test.ppm', '3.png']
+        images = [os.path.join(tmpdir, f) for f in filenames]
+
+        class1 = ['test.png', 'test0.jpg', '1.jpeg']
+        images += [os.path.join(tmpdir, 'class1', f) for f in class1]
+
+        class2 = ['test.bmp', 'test0.ppm', '1.jpeg']
+        images += [os.path.join(tmpdir, 'class2', f) for f in class2]
```

Comment:
no need for all that boiler code. `images = [os.path.join(tmpdir, subdir, f) for subdir in ['', 'subdir1', 'subdir2'] for f in filenames]`